src/extensions/shared/paste-spreadsheet-table.ts

```diff
@@ -0,0 +1,74 @@
+import { Extension } from '@tiptap/core'
+import { Plugin, PluginKey } from 'prosemirror-state'
+
+import { PASTE_EXTENSION_PRIORITY } from '../../constants/extension-priorities'
+import { parseHtmlToElement } from '../../helpers/dom'
+
+/**
+ * The `PasteSpreadsheetTable` extension adds the ability to paste a table copied from a spreadsheet
+ * web app (e.g., Google Sheets, Microsoft Excel), along with tables rendered by GitHub Flavored
+ * Markdown (GFM), into the editor.
+ *
+ * Since Typist does not yet support tables, this extension simply pastes the table as a string of
```

Comment:
Do you think it would be helpful to be more explicit in the extension's name (e.g. pasteSpreadsheetTableAsParagraphs)? It might be easier to figure out at a glance what's supported, but if we ever do support markdown tables in the future, then we could reuse this name while re-implementing its internals.

Reply:
I renamed the extension to be more explicit on what it does.

However, it got me thinking about this comment that I wrote, which doesn't make much sense... If we ever add proper table suppor to the editor, this extension **should not** be rewritten as I first through, because we are still gonna need the functionality that it provides for editors where tables aren't supported (i.e. Todoist task title). Instead, what we need to do is either enable/disable this extension when required.

With that in mind, I'm thinking if it wouldn't make sense to revert the way it's included in the rich-text/plain-text kits. I mean, right now I added them by default to both kits, and we can disable them per editor. Would it make sense to have this extension not added by default to both kits, and with the option enable them per editor?

Also, we don't necessarily need to have it enable/disable by default on both the rich-text/plain-text (one can be enabled by defualt while we disable by default on the other), because for rich-text we might add table support down the road and have it turned on by default, but for plain-text we'll never do that, because, well, it's plain-text (at most we'd need a new extension that would transform an HTML table into a Markdown one, but that's a different feature altogether).

What do you think? Does it make sense that such extension is enabled by default or considered an extra that we enable when we need it? Should we have the same behaviour for both kits, or different?

Reply:
IMO it makes sense to enable this by default. I can't think of a use case where users would want to paste in spreadsheets cells that they've copied and have them all mashed together. We only use the single-line plain text kit in Twist, and I think in that case it's still better for the spaces to appear between cells.

When/if we do support markdown tables, I think this would still be a nice fallback when it's disabled. So my vote would be to have this enabled by default now for both kits 👍